### DISCUSSION TITLE
BigTIFF support ?

### DISCUSSION THREAD

**Discussion by geektortoise** [ngageoint/tiff-java#22] [Ideas]
#### Version Information:

  * TIFF Java Version: 2.0.0
  * Java Version: (java -version) : openjdk version "1.8.0_181"
  * Platform & OS: Debian

#### Expected Results:

  * What did you expect to happen? Can use the librairy to get metadata of a bigTIFF

#### Observed Results:

  * What happened instead? 
mil.nga.tiff.util.TiffException: Invalid file identifier, not a TIFF

  * How often does this occur? Every time

#### Output:

  * Any logs, errors, or output messages?
mil.nga.tiff.util.TiffException: Invalid file identifier, not a TIFF

#### Steps to Reproduce:

  1. TIFFImage tiffImage = TiffReader.readTiff(new File("/tmp/test.tif"));
where test.tif is a bigtiff (so the 
https://github.com/ngageoint/tiff-java/blob/master/src/main/java/mil/nga/tiff/util/TiffConstants.java#L23
FILE IDENTIFIER is 43)

#### Test Files:

  * Links to any files needed for testing? you can convert a jpeg to bigtiff with libvips
/usr/local/bin/vips tiffsave "test.jpeg" "test.tif" --tile --pyramid  --bigtiff

I know that it is not possible for java to handle a huge file > 2Gb or 4 Gb. But is it planned to be able to read and modify metadata of bigtiff ?

**Comment by bosborn** [comment]
Nothing currently planned, but not opposed to supporting and open to pull requests.

**Comment by Samyssmile** [comment]
> #### Version Information:
> * TIFF Java Version: 2.0.0
> * Java Version: (java -version) : openjdk version "1.8.0_181"
> * Platform & OS: Debian
> 
> #### Expected Results:
> * What did you expect to happen? Can use the librairy to get metadata of a bigTIFF
> 
> #### Observed Results:
> * What happened instead?
>   mil.nga.tiff.util.TiffException: Invalid file identifier, not a TIFF
> * How often does this occur? Every time
> 
> #### Output:
> * Any logs, errors, or output messages?
>   mil.nga.tiff.util.TiffException: Invalid file identifier, not a TIFF
> 
> #### Steps to Reproduce:
> 1. TIFFImage tiffImage = TiffReader.readTiff(new File("/tmp/test.tif"));
>    where test.tif is a bigtiff (so the
>    https://github.com/ngageoint/tiff-java/blob/master/src/main/java/mil/nga/tiff/util/TiffConstants.java#L23
>    FILE IDENTIFIER is 43)
> 
> #### Test Files:
> * Links to any files needed for testing? you can convert a jpeg to bigtiff with libvips
>   /usr/local/bin/vips tiffsave "test.jpeg" "test.tif" --tile --pyramid  --bigtiff
> 
> I know that it is not possible for java to handle a huge file > 2Gb or 4 Gb. But is it planned to be able to read and modify metadata of bigtiff ?

Did you found another way to open BigTiff?

**Reply by Samyssmile** [reply]
Hi, I wrote my own software fot that, but dont have code anymore

**Comment by bosborn** [comment]
_Converted from an Issue to a Discussion_

**Comment by rdkgit** [comment]
We would love support for BigTIFF as well.
